### ISSUE TITLE
A request for Docker Image

### ISSUE THREAD

**Issue by cpu** [jlund/streisand-discussions#20]
Suggested by @aminbolehvpn in https://github.com/jlund/streisand/issues/126

> Hi Joshua,
> Is it possible to have docker images? It will do the deployment of streisand a lot quicker to novices.

There was some follow-up discussion in https://github.com/jlund/streisand/issues/126 including links to `docker-streisand` projects others have put together that should be referenced by anyone wishing to restart this discussion. 

Also an older PR: https://github.com/jlund/streisand/pull/442

**Comment by ronnicek** [comment]
Hi,

I was thinking about how to keep Streisand up-to-date and it could be great to have some core script in shell on system and then all services in docker images (one service per docker image) and you can simply rebuild dockerimages to have fresh system with all features (also new features). Something like https://cloudron.io.

Because have everything in one docker image is against conteiner philosophy.

**Comment by EduardoRT** [comment]
Indeed, it'll be pretty cool to have a docker image / docker-compose file for the whole project. I've got this habit of killing my Streisand servers and then recreating them from scratch. Having a Docker image/docker-compose.yaml will make this process even easier/faster (it's really fast now, I have to admit)

This will also remove the need of a specific Ubuntu version and it won't pollute the host system.

**Comment by nopdotcom** [comment]
We are always looking for PRs. I'm not a docker expert.

**Comment by jamesmstone** [comment]
It would be significantly easier to create a docker image, if the setup script was able to run non interactive, taking setup q's from either cli arguments, env variables or a config file

**Comment by cpu** [comment]
@jamesmstone I opened https://github.com/jlund/streisand/issues/972 for this suggestion. Unfortunately it won't be trivial to make this happen with the current Streisand codebase and its set of genesis roles.

**Comment by cpu** [comment]
@jamesmstone There's now support for [a non-interactive deployment mode](https://github.com/StreisandEffect/streisand/commit/c9339a6d2ad08dc933cb41afd73fb0162972f95b) in master.
